### ISSUE TITLE
config: allow setting the MD data directory

### ISSUE THREAD

**Issue by smashah** [open-wa/wa-automate-nodejs#2311]

**Comment by smashah** [comment]
Here's how it will resolve in order:
1. userDataDir (the whole path)
2. if in docker then default to /sessions/_IGNORE_XYZ
3. if not in docker then fallback to {mdDir}/_IGNORE_XYZ
4. if no mdDir is set then use the process working directory

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}
//set the version type
process.env.VERS = "patch"

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i -D`);
exec(`npm run release-ci $VERS`);

// types only package
exec('npm config set //registry.npmjs.org/:_authToken $NPM_TOKEN')
exec(`cd ./types-only && npm version $VERS && npm run build && npm publish && cd ..`);
exec(`git commit -a -m 'updated types-only package'`);
exec(`git push --force`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [4.23.15](https://github.com/open-wa/wa-automate-nodejs/compare/4.23.14...4.23.15) (2021-11-20)

- Release 4.23.15 [`67428ad`](https://github.com/open-wa/wa-automate-nodejs/commit/67428ad1808d0a251b11e8f3f93cfb67056f0b4c)
- 📝  fix readme [`baef3fc`](https://github.com/open-wa/wa-automate-nodejs/commit/baef3fc6fac96f047fb49ce8a0e373b494fb0f02)
- 🐛fix `config.devtools` #2310 [`b580a17`](https://github.com/open-wa/wa-automate-nodejs/commit/b580a178ed2eae902d8c74fb4227c97bac245bc5)
- ⬆️ upgrade devtools dependency #2310 [`b4fc385`](https://github.com/open-wa/wa-automate-nodejs/commit/b4fc385e77419e8d5092b04856a926af46318401)
- 🩹 fix MD auto-detection for custom session data paths #2311 [`7b8ce64`](https://github.com/open-wa/wa-automate-nodejs/commit/7b8ce64b2a9fd479275cbfde0b76abd36c122fea)
- ✨ config: set custom paths for MD directories #2311 [`15708b5`](https://github.com/open-wa/wa-automate-nodejs/commit/15708b5f883ccf31fdf9a7c0f5ae5a1f9f46b69d)
- 🥚 patches update [`78ff21d`](https://github.com/open-wa/wa-automate-nodejs/commit/78ff21d4f347a98556b73dc6169a4bd1ae63380f)
- 🥚 patches update [`f63bcb2`](https://github.com/open-wa/wa-automate-nodejs/commit/f63bcb2f6a8ba9677d2bf0d2cffca5b581a5f3e2)
- updated types-only package [`0822c9b`](https://github.com/open-wa/wa-automate-nodejs/commit/0822c9b872dbcd7b64ab86f8872413a066217ea5)
